spec/lib/rabbit_feed/event_spec.rb

```diff
@@ -67,6 +67,65 @@ module RabbitFeed
       end
     end
 
+    describe '#serialize' do
+      let(:schema) do
+        Avro::Schema.parse(
+          {
+            name:   'example-1.0',
+            type:   'record',
+            fields: [
+              {
+                name: 'payload',
+                type: {
+                  name: 'event_payload',
+                  type: 'record',
+                  fields: [
+                    { name: 'event_integer', type: 'int' },
+                    { name: 'event_string',  type: 'string' },
+                  ],
+                },
+              },
+              {
+                name: 'metadata',
+                type: {
+                  name: 'event_metadata',
+                  type: 'record',
+                  fields: [{ name: 'name', type: 'string' }],
+                },
+              },
+            ],
+          }.to_json
+        )
+      end
+
+      context 'with invalid payload' do
+        let(:payload)  { {
+          'event_string'  => 'HIGHLY SENSITIVE',
+          'event_integer' => 'incorrect',
+        } }
+
+        it 'raises an Exception' do
+          expect {
+            subject.serialize
+          }.to raise_error(Avro::IO::AvroTypeError)
+        end
+
+        it 'can remove values from exception' do
+          event = described_class.new(metadata, payload, schema, ['event_string'])
+          exception_msg = nil
+          begin
+            event.serialize
+          rescue Avro::IO::AvroTypeError => e
+            exception_msg = e.message
+          end
+          expect(exception_msg).to_not be_nil
+          expect(exception_msg).to_not include("INCORRECT")
```

Comment:
That was my guess, too. Methinks someone might not have started with a failing test :-P